plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -165,7 +166,7 @@ private Stream<DeltaLakeSplit> getSplits(
                         return Stream.empty();
                     }
 
-                    if (filesModifiedAfter.isPresent() && addAction.getModificationTime() <= filesModifiedAfter.get().toEpochMilli()) {
+                    if ((!ignoreOldStats.orElse(false)) && recalculateAllStatistics.isPresent() && addAction.getModificationTime() <= recalculateAllStatistics.get().toEpochMilli()) {
```

Comment:
There should be no need to check ignoreOldStats. filesModifiedAfter should is empty for recalculation.

Reply:
Yeah. This part doesn't seem right, besides the parenthesis around the first item being redundant. 
```
(!ignoreOldStats.orElse(false)) && recalculateAllStatistics.isPresent()
```